### PR TITLE
fix(guardrails): require write signal for computed call write_bypass

### LINKED ISSUE #2722: guardrails: ps::write_bypass reports any computed call target as a file write, with no write indicator required
## Problem

`ps::write_bypass` reports any call `&` or dot-source `.` of a **computed target** as a
file-write-cmdlet bypass, with no write indicator required anywhere in the command. So commands
that author no content at all are blocked with a message stating they author content.

`plugins/guardrails/lib/powershell/ps-command.sh` (current `main`):

```bash
# A call/dot-source of a COMPUTED target — `& ('Set-'+'Content') …`, `& $w …`
# — evaluates an expression into the command name; it cannot be proven
# non-writer, so it fails closed like iex (review round 5). …
if [[ "$lcq" =~ (^|[[:space:]\;\{\}\(\|\&])[.\&][[:space:]]*[\(\$] ]]; then
  return 0
fi
```

The predicate is the invocation *shape* alone. Unlike the git lane — where
`ps::classify_git_command` reaches its computed-target test only after
`ps::has_special_constructs` has already routed the command to the fail-closed sink — this branch
has **no precondition**: no special construct, no writer name, no redirect, no `-Value`, nothing.

## Measured

A harness sourcing the library and calling `ps::write_bypass` directly, against `origin/main`:

| command | verdict | authors a file? |
|---|---|---|
| `. $PROFILE` | **write** | no |
| `& $py script.py` | **write** | no |
| `& $tool --version` | **write** | no |
| `. $env:MY_PROFILE` | **write** | no |
| `& $python -m unittest discover` | **write** | no |

All five are reported as write bypasses. For contrast, every genuine case still classifies
correctly — `Set-Content`, `Add-Content`, `Out-File`, `Write-Output > f`, `New-Item -Value`,
`[IO.File]::WriteAllText`, `iex $payload`, and `& 'Set-Content'` — as does ordinary read-only work
(`Get-ChildItem`, `git status`, `Get-Content`). 11 pass, 5 fail.

So this is a clean over-block, not a precision/recall tradeoff: narrowing it does not require
giving up any true positive in the harness.

## Why it matters operationally

`. $PROFILE` is the standard way to reload a PowerShell profile. `& $interpreter script.py` is the
standard way to invoke a pinned interpreter held in a variable — and it is **precisely the shape a
sibling plugin's denial message instructs operators to use**: `disk-hygiene`'s Bash-belt message
tells the operator to run the engine "using the hook's absolute Python interpreter", and binding
that long path to a variable is the idiomatic PowerShell way to satisfy it.

The block message compounds it. The text names file-write cmdlets and redirection, and its own
Scope paragraph enumerates a covered surface — content-authoring cmdlets, `New-Item -Value`, the
`Export-*` family, .NET file writes, producer-scoped redirects, `iex`. A computed call target
appears nowhere in that list, so an operator reading the message cannot tell what tripped it, and
the natural remediations (remove the redirect, use `Write` instead) do not apply.

## Test coverage

The shipped `plugins/guardrails/hooks/block-hook-bypass.test.sh` (~78 KB) contains **zero** cases
of this shape:

```console
$ grep -c -F '& $' plugins/guardrails/hooks/block-hook-bypass.test.sh
0
```

So the behavior is untested rather than test-defended, in either direction — nothing pins the
current over-block, and nothing would catch a narrowing that went too far.

## Suggested direction

The comment justifies failing closed on the grounds that a computed target "cannot be proven
non-writer". That reasoning is sound for the *sink*; the question is whether the shape alone should
enter the sink with no other signal present.

Two options, both cheap to evaluate against the harness above:

1. **Add a precondition**, mirroring the git lane: require a write indicator OR a special construct
   to be present before the computed-target shape decides. A bare `& $py script.py` with no
   redirect, no writer name, and no `-Value` would then pass.
2. **Keep failing closed but fix the message** — name the computed call target as the trigger and
   give an actionable remediation (`& "<literal path>"`), the way the git lane's
   `ps::print_sink_trigger_line` distinguishes its four sink shapes. #2662 did exactly this for the
   git lane's messages.

Whichever is chosen, add regression cases for the five commands above and for the eight genuine
bypasses, so the boundary is pinned in both directions.

## Related

- #2662 — corrected the PowerShell fail-closed *messages* not to assert a git command is present;
  the same message-accuracy problem exists here for writes.
- #2667 — reworked this file (lazy classifier load, sink-shape allow-list); this branch was not
  changed by it.
- #1968 — the precedent for narrowing a computed-target test: the git lane stopped treating a
  constant literal target (`& "publish.ps1"`) as computed.
- #2720 — a separate correctness fix in the same file (`ps::git_command_is_readonly`), open now.

No linked issue: this report opens the issue rather than closing one.

### PR DESCRIPTION
**Pull request by kyle-sexton**
Closes #2722

## Summary

`ps::write_bypass` was treating any PowerShell call/dot-source of a computed target (`& $tool …`, `. $PROFILE`) as a file-write bypass with no write indicator required, so non-writing commands were blocked with a message that named writers and redirects.

## Fix

Gate the computed-target fail-closed probe on a write signal first — a special construct (`()`/`{}`/backtick/`--%`), a producer redirect (`>` after fd-dup merges are stripped), or a `-Value`/`-va*` parameter — mirroring the git lane where `ps::might_invoke_git`'s computed-target test only runs after a sink trigger. Bare `& $tool …` / `. $PROFILE` become the same deferred residual `has_dynamic_invocation` already documents. Genuine writers (`Set-Content`, `& 'Set-Content'`, `& ('Set-'+'Content') …`, `& $w -Value …`, `& $w > f`, `iex`, .NET writes) still block. Plugin bumped to `0.28.28` with a CHANGELOG entry.

## Verification

- `bash plugins/guardrails/hooks/block-hook-bypass.test.sh` — `PASS=434 FAIL=0` (includes five #2722 allow cases plus `-Value` / redirect / `2>&1` boundary cases)
- Direct `ps::write_bypass` harness — 20/20 on the issue's allow and genuine-write matrix
- `bash scripts/check-changelog-parity.sh --check-bump origin/main` — pass

## Related

Refs #2662 (same message-accuracy class on the git lane), Refs #1968 (computed-target narrowing precedent), Refs #2667 (ps-command sink-shape work that left this branch unchanged)

### CODE REVIEW

File: plugins/guardrails/lib/powershell/ps-command.sh [hunk 1]
```diff
@@ -1013,12 +1017,24 @@ ps::write_bypass() {
...
-  if [[ "$lcq" =~ (^|[[:space:]\;\{\}\(\|\&])[.\&][[:space:]]*[\(\$] ]]; then
-    return 0
+  if ps::call_target_is_bare_computed "$lcq"; then
+    blanked_gate=$(ps::blank_quoted_spans "$lcq")
+    # fd-dup merges (`2>&1`) are plumbing, not file writes — strip before the
+    # redirect probe so `& $tool 2>&1` does not look like a producer redirect.
+    gate=$(printf '%s' "$lcq" | sed -E 's/[0-9*]*>&[0-9]+//g')
+    if ps::has_special_constructs "$blanked_gate" ||
+      [[ "$gate" == *'>'* ]] ||
+      [[ "$lcq" =~ [[:space:]]-va[a-z]*([[:space:]]|:) ]]; then
```
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/guardrails/lib/powershell/ps-command.sh]: Addressed: quote-blanked redirect/-va* probes; positional Path+Value (two leading non-flag tokens) and pipeline+path shapes now fail closed. Regression cases added; suite PASS=438.

File: plugins/guardrails/lib/powershell/ps-command.sh [hunk 2]
```diff
@@ -1013,12 +1017,24 @@ ps::write_bypass() {
...
+      return 0
+    fi
   fi
```
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/guardrails/lib/powershell/ps-command.sh:1085]: Addressed: quote-blanked redirect/-va* probes; positional Path+Value (two leading non-flag tokens) and pipeline+path shapes now fail closed. Regression cases added; suite PASS=438.

File: plugins/guardrails/lib/powershell/ps-command.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/guardrails/lib/powershell/ps-command.sh]: Addressed: quote-blanked redirect/-va* probes; positional Path+Value (two leading non-flag tokens) and pipeline+path shapes now fail closed. Regression cases added; suite PASS=438.
